cordapp-upgrades/README.md

```diff
@@ -0,0 +1,158 @@
+ `./gradlew issueBetweenNodes`, which will issue an obligation between each pair of nodes in the network).
+ 
+2. Shut down the network, then upgrade one of the nodes to the new version of the contract. This also requires
+ upgrading to the latest version of the workflows CorDapp, as older versions are compiled against the initial contract.
+ The upgrade can be carried out by running `scripts/upgradeNodes.sh workflows v3-can-default PartyA`
+ and `scripts/upgradeNodes.sh contracts v2-can-default PartyA`. Note: There is some additional configuration for the nodes running the new versions of the workflows CorDapp. This
+ configuration is used to control whether the new field in the contract state is set when the state is created. The no
+ downgrade rule will prevent nodes with only the old contract version from using the state if it has new fields set, and
+ so as a result the new field cannot be set until all nodes in the network are upgraded. To begin with, this configuration
+ prevents the field from being populated.
+ 
+3. Run the network (`build/nodes/runnodes`). In order to issue obligations between nodes, the contract attachment must be
+ present in the attachment store of each node (if this isn't the case, then an error will be hit when sending a transaction
+ with a version of the contract the node doesn't recognise). To put a copy of each contract version in the network in each
+ node's attachment store, run `./gradlew loadAttachments`. (This command opens all attachments on all nodes, and then
+ uploads each attachment to each node that doesn't already have it.)
+ 
+ 4. Check that obligations can be issued between nodes (again, `./gradlew issueBetweenNodes` can be used). It is
+ instructive to show that obligations can still be freely transferred between nodes running different contract versions
+ at this point. To do this, first clear all obligations from the nodes (the easiest way to do this is to settle them all
+ using `./gradlew settleAllObligations`), then do the following:
+  
+   - Issue an obligation from the shell of PartyB, naming PartyA as the lender (`start IssueObligation$Initiator amount:
+   $100, lender: PartyA, anonymous: true`)
+   - On PartyA, do `run vaultQuery contractStateType: net.corda.examples.obligation.contract.Obligation` and note the
+   linear ID of the obligation state.
+   - Transfer the obligation to PartyC: `start TransferObligation$Initiator newLender: PartyC, linearId: <linear ID>, anonymous: true`
+   - Transfer the obligation back, by running the same command on PartyC and naming PartyA as the new lender.
+  
+5. Shut down the network, then upgrade the final node (PartyC) to use the new contract:
+   - `scripts/upgradeNodes.sh workflows v3-can-default PartyB PartyC`
+   - `scripts/upgradeNodes.sh contracts v2-can-default PartyB PartyC`
+  
+6. Adjust the CorDapp configuration to set the new field in the obligation state. (Note that in a real system, this
```

Comment:
I don't understand this configuration field. Surely V3 of the Flow will use/set this (as it has been designed to do so).
I recommend not using a configuration field to avoid mis-directing end users (remember, they tend to cut'n'paste sample code).